src/App.tsx

```diff
@@ -17,18 +53,28 @@
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                status={status}
+                query={query}
+                setStatus={setStatus}
+                setQuery={setQuery}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading && <Loader />}
+              <TodoList
+                todos={filteredTodos}
+                selectedTodo={selectedTodo}
+                onSelectedTodo={setSelectedTodo}
+              />
             </div>
           </div>
         </div>
       </div>
-
-      <TodoModal />
+      {!!selectedTodo && (
```

Comment:
You are directly interacting with the DOM here by using the `&&` operator to conditionally render the `TodoModal` component. It's recommended to avoid direct DOM manipulations in React. Instead, you could use conditional rendering in a more 'React' way. For example, you could use a ternary operator to render the `TodoModal` component or `null` based on whether `selectedTodo` is truthy.